CMakeLists.txt

```diff
@@ -542,58 +542,59 @@ else()
 endif()
 
 if(NOT MSVC)
-  find_program(MOLD_EXECUTABLE mold)
-  default_option(MOLD_SUPPORT "Use 'mold' for linking" "MOLD_EXECUTABLE")
+  # Try to configure mold as linker via -fuse-ld=mold, support from gcc 12.1, gcc 11.2.0-16 or clang
+  execute_process(
```

Comment:
Maybe use https://cmake.org/cmake/help/latest/module/CheckCompilerFlag.html?

Reply:
The advantage of querying the version via the compiler is that the whole chain is checked and not only the compiler flag. 
check_compiler_flag() does only check the flag.